src/product-category/product-category.service.ts

```
@@ -0,0 +1,135 @@
+   * @returns {Promise<ProductCategory>}
+   * @throws {NotFoundException, ConflictException}
+   */
+  async update(id: number, updateProductCategoryInput: UpdateProductCategoryInput): Promise<ProductCategory> {
+    const productCategory = await this.productCategoryRepository.findOne({ where: { id } });
+
+    if (!productCategory) {
+      throw new NotFoundException(`Product Category #${id} not found`);
+    }
+
+    if (updateProductCategoryInput.name) {
+      productCategory.name = updateProductCategoryInput.name;
+    }
+
+    if (updateProductCategoryInput.description) {
+      productCategory.description = updateProductCategoryInput.description;
+    }
+
+    try {
+      return await this.productCategoryRepository.save(productCategory);
+    } catch (error) {
+      if (error.code === '23505') {
+        // Handle the unique_violation error
+        throw new ConflictException('Name already exists');
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Remove a product category, based on id
+   * ONLY remove if not associated product(s)
+   *
+   * @param {number} id
+   * @returns {Promise<{ message: string }>}
+   * @throws {NotFoundException, ConflictException, InternalServerErrorException}
+   */
+  async remove(id: number): Promise<{ message: string }> {
+    const productCategory = await this.productCategoryRepository.findOne({ where: { id } });
+
+    if (!productCategory) {
+      throw new NotFoundException(`Product Category with ID #${id} not found`);
+    }
+
+    // Check for products associated with the supplier
+    const associatedProducts = await this.productRepository.count({ where: { category: productCategory } });
+
+    if (associatedProducts > 0) {
+      throw new ConflictException(`Cannot delete product cagetory with ID #${id} as it has associated products.`);
+    }
+
+    try {
+      await this.productCategoryRepository.remove(productCategory);
+    } catch (error) {
```

We should never "swallow" errors. We should at least log them. Imagine that you're faced with a defect about this and you're asked to find the root cause - you would have no logs with the actual DB exception...